src/main/java/uk/ac/cam/cl/dtg/segue/dao/kafkaStreams/SiteStatisticsStreamsApplication.java

```diff
@@ -160,13 +164,28 @@ public SiteStatisticsStreamsApplication(final PropertiesLoader globalProperties,
 
         // use the builder and the streams configuration we set to setup and start a streams object
         streams = new KafkaStreams(builder, streamsConfiguration);
+
+        // handling fatal streams app exceptions
+        streams.setUncaughtExceptionHandler(
+                (thread, throwable) -> {
+
+                    // a bit hacky, but we know that the rebalance-inducing app death is caused by a CommitFailedException that is two levels deep into the stack trace
+                    // otherwsie we get a StreamsException which is too general
+                    if (throwable.getCause().getCause() instanceof CommitFailedException) {
+                        streamThreadRunning = false;
```

Comment:
Could we write to the log to let us know that this is happening?

Reply:
I'll change this now